pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/PendingAckHandleImpl.java

```diff
@@ -155,13 +155,13 @@ public PendingAckHandleImpl(PersistentSubscription persistentSubscription) {
                         .getBrokerService().getPulsar().getTransactionPendingAckStoreProvider();
 
         pendingAckStoreProvider.checkInitializedBefore(persistentSubscription)
-                .thenAccept(init -> {
+                .thenAcceptAsync(init -> {
                     if (init) {
                         initPendingAckStore();
                     } else {
                         completeHandleFuture();
                     }
-                })
+                }, internalPinnedExecutor)
```

Comment:
Out of curiosity, how did you select this thread to run the task?

Reply:
Every transactions related tasks run in the executor managed by `numTransactionReplayThreadPoolSize` property.
Since we don't want to waste the metadata-store thread pool, this one sounds the more appropriate. It's also the same one used for other tasks in this class

Reply:
Thanks for the explanation.